### ISSUE TITLE
Support single .csproj in multiple .sln files

### ISSUE THREAD

**Issue by Bosch-Eli-Black** [SonarSource/sonarlint-visualstudio#1788]
Our team uses multiple .sln files, and several of those .sln files include the same .csproj project file, with the result being that the .csproj file is included in multiple different .sln files.

For example:

```
App.sln
--App.csproj
--SharedLibrary.csproj

Website.sln
--Website.csproj
--SharedLibrary.csproj
```

When we bind App.sln to our SonarQube project, everything goes smoothly. However, when we then try to bind Website.sln to our SonarQube project, we get an error that says this:

```
Unexpected error during workflow execution: Failed to bind project '....\Website\SharedLibrary.csproj'.
A conflicting version of SonarLint.xml has been found. Please delete the file '....\App\.sonarlint\....\CSharp\SonarLint.xml' and remove references to it from your projects, then re-open the solution and try again..
```

It would be great if SonarLint could support this use case 🙂  (Or does it already, and I just need to configure something?)

Also, it would be great if SonarLint supported having the .sonarlint directory be outside of out the solution directory. Ideally, our above setup would look something like this, with the projects in App.sln and Website.sln referencing the same to-level .sonarlint directory:

```
.sonarlint
App.sln
Website.sln
```

Thanks 🙂

**Comment by rita-g-sonarsource** [comment]
Hi @Bosch-Eli-Black.

We don't support binding the same MSBuild project in multiple solutions but there is a workaround to your use case: you can completely exclude `SharedLibrary.csproj` from binding by adding `SonarQubeExclude=true` property in the csproj. 

This will cause SonarLint to ignore `SharedLibrary.csproj` during the binding and your binding should succeed. You can then manually modify `SharedLibrary.csproj` and [specify the ruleset files](https://github.com/SonarSource/sonarlint-visualstudio/wiki/Connected-Mode-configuration-for--C%23-and-Visual-Basic-projects) that you want to use (either from Website.sln or App.sln). In fact, if the project is already bound, then those properties will already be set and you could just leave them as-is and just add the `SonarQubeExclude` property. 

Please feel free to submit a feature request at our [community forum](https://community.sonarsource.com/) -- it's a better place for suggestions/questions, as it has higher visibility and a voting system for prioritization. For now we will close this ticket.

**Comment by Bosch-Eli-Black** [comment]
@rita-g-sonarsource  Thanks for the quick and helpful reply! 🙂  I've filed a feature request [here](https://community.sonarsource.com/t/support-including-csproj-in-multiple-sln-files-visual-studio-sonarlint/32885).

My understanding is that this workaround would make it so that `SharedLibrary.csproj` would generate warnings (because of the `.ruleset` file) but the warnings wouldn't be synced with SonarQube, so if we had a warning in `SharedLibrary.csproj` that we marked as "won't fix" in SonarQube, the warning would still show up in Visual Studio. Is that correct?

Thanks again 🙂

**Comment by rita-g-sonarsource** [comment]
@Bosch-Eli-Black it should work, are you experiencing issues with it? If you're using VS 2019, we have a known issue that suppressions are not handled well - #1005.

**Comment by Bosch-Eli-Black** [comment]
@rita-g-sonarsource Thanks again for the reply. I hadn't yet tried your suggestion; I was just trying to mentally work through the implications of setting `SonarQubeExclude=true` 🙂

Anyway, I've now tried the steps that you suggested. As you said, this lets us successfully bind both the `App.sln` and `Website.sln` solutions. This also lets us see warnings in Visual Studio for code in `SharedLibrary.csproj`. For example, if I add some commented out code to `SharedLibrary.csproj`, Visual Studio shows a warning.

However, we also scan our code via Azure DevOps every time we commit, and it looks like adding the `SonarQubeExclude=true` makes it so that `SharedLibrary.csproj` doesn't get scanned, so the warnings that we see in Visual Studio aren't reflected in our SonarQube server.

Is there any way to work around that issue?

Thanks again 🙂

**Comment by rita-g-sonarsource** [comment]
Hi @Bosch-Eli-Black, you can get around this by wrapping `SonarQubeExclude` with a condition that only exists on local builds / doesn't exist on the CI build (e.g. Debug configuration, MSBuild property $(BuildingInsideVisualStudio), AzureDevops agent properties, etc).

**Comment by Bosch-Eli-Black** [comment]
Thanks, @rita-g-sonarsource! That worked 🙂
